influxdb3_catalog/src/catalog.rs

```diff
@@ -1620,51 +1597,4 @@ mod tests {
             .expect_err("should fail to apply AddFields operation for non-existent table");
         assert_contains!(err.to_string(), "Table banana not in DB schema for foo");
     }
-
-    #[test]
-    fn test_check_and_mark_table_as_deleted() {
-        let db_id = DbId::new();
-        let deleted_table_id = TableId::new();
-        let table_name = Arc::from("boo");
-        let deleted_table_defn = DeleteTableDefinition {
-            database_id: db_id,
-            database_name: Arc::from("foo"),
-            table_id: deleted_table_id,
-            table_name: Arc::clone(&table_name),
-            deletion_time: 0,
-        };
-        let mut map = IndexMap::new();
-        let table_defn = Arc::new(
-            TableDefinition::new(
-                deleted_table_id,
-                Arc::clone(&table_name),
-                vec![
-                    (ColumnId::from(0), "tag_1".into(), InfluxColumnType::Tag),
-                    (ColumnId::from(1), "tag_2".into(), InfluxColumnType::Tag),
-                    (ColumnId::from(2), "tag_3".into(), InfluxColumnType::Tag),
-                    (
-                        ColumnId::from(3),
-                        "time".into(),
-                        InfluxColumnType::Timestamp,
-                    ),
-                    (
-                        ColumnId::from(4),
-                        "field".into(),
-                        InfluxColumnType::Field(InfluxFieldType::String),
-                    ),
-                ],
-                vec![ColumnId::from(0), ColumnId::from(1), ColumnId::from(2)],
-            )
-            .unwrap(),
-        );
-        map.insert(deleted_table_id, table_defn);
-        let mut updated_or_new_tables = SerdeVecMap::from(map);
-
-        check_and_mark_table_as_deleted(Some(&deleted_table_defn), &mut updated_or_new_tables);
-
-        let deleted_table = updated_or_new_tables.get(&deleted_table_id).unwrap();
-        assert_eq!(&*deleted_table.table_name, "boo-19700101T000000");
-        assert!(deleted_table.deleted);
-        assert!(!deleted_table.series_key.is_empty());
-    }
```

Comment:
There should be a new test added in this one's place, unless you had justification for removing it, but that is not clear.

I think you can replicate this with the new APIs you've added. That is my only recommendation for this PR.

Reply:
Initially deleted it as it was testing a function that I'd removed, `check_and_mark_table_as_deleted`. I'll add an equivalent test that validates the CatalogOp does the correct thing.

Reply:
Updated the commit

Reply:
Looks good!